### PR TITLE
Add library for communication with the motor controller

### PR DESCRIPTION
**Pull request by andebjor**
This commit also adds a "fake" implementation of the required UART abstraction needed by the third party library. That is, an implementation that has the correct API for building with, but that won't be useful at runtime.

### CODE REVIEW

File: .bazelrc [hunk 1]
```diff
@@ -7,19 +7,22 @@ build:gcc12   --extra_toolchains=//toolchain:gcc12
...
 build:clang-format --@bazel_clang_format//:binary=@llvm_16_toolchain//:clang-format
 build:clang-format --@bazel_clang_format//:config=//:format_config
+build:clang-format --@bazel_clang_format//:ignore=//:format_ignore
 build:clang-format --output_groups=report
 build:clang-format --keep_going
 
 build:clang-tidy-base --config=clang16
 build:clang-tidy-base --aspects @bazel_clang_tidy//clang_tidy:clang_tidy.bzl%clang_tidy_aspect
-build:clang-tidy-base --@bazel_clang_tidy//:clang_tidy_config=//:tidy_config
+#build:clang-tidy-base --@bazel_clang_tidy//:clang_tidy_config=//:tidy_config
```
Comment (oliverlee) [inline .bazelrc]: this can be removed

File: .bazelrc [hunk 2]
```diff
@@ -7,19 +7,22 @@ build:gcc12   --extra_toolchains=//toolchain:gcc12
...
 build:clang-tidy-base --output_groups=report
 build:clang-tidy-base --keep_going
+build:clang-tidy-base --spawn_strategy=local
 
 build:verbose-clang-tidy --config=clang-tidy-base
 build:verbose-clang-tidy --@bazel_clang_tidy//:clang_tidy_executable=//tools:verbose-clang-tidy
 
 build:clang-tidy --config=clang-tidy-base
 build:clang-tidy --@bazel_clang_tidy//:clang_tidy_executable=@llvm_16_toolchain//:clang-tidy
+build:clang-tidy --spawn_strategy=local
```
Comment (oliverlee) [inline .bazelrc]: this shouldn't be necessary

File: third_party/.clang-tidy [hunk 3]
```diff
@@ -0,0 +1,9 @@
+Checks: >
```
Comment (oliverlee) [inline third_party/.clang-tidy]: we can probably delete this?

File: third_party/vesc_uart/BUILD.bazel [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        "WProgram.h",
+        "buffer.cpp",
+        "buffer.h",
+        "crc.cpp",
+        "crc.h",
+        "datatypes.h",
+        "local_datatypes.h",
+    ],
+    hdrs = ["VescUart.h"],
+    copts = [
```
Comment (oliverlee) [inline third_party/vesc_uart/BUILD.bazel:16]: some of these options may not be necessary with use of  `includes`

and we'll probably want to include as a system header so warnings don't need to be suppressed by downstream targets

File: uart_abstraction/BUILD.bazel [hunk 5]
```diff
@@ -0,0 +1,11 @@
+load("@rules_cc//cc:defs.bzl", "cc_library")
+
+cc_library(
+    name = "fake",
+    srcs = [
+        "fake/HardwareSerial.cpp",
+    ],
+    hdrs = ["fake/HardwareSerial.hpp"],
```
Comment (oliverlee) [inline uart_abstraction/BUILD.bazel]: Will this header eventually be shared by multiple implementations?

File: third_party/.clang-tidy (hunk 3 above)
Comment (andebjor) [reply third_party/.clang-tidy]: Yes, works locally without it.

File: third_party/vesc_uart/BUILD.bazel (hunk 4 above)
Comment (andebjor) [reply third_party/vesc_uart/BUILD.bazel:16]: Yep. I have restructured the code a bit

File: uart_abstraction/BUILD.bazel (hunk 5 above)
Comment (andebjor) [reply uart_abstraction/BUILD.bazel]: Good point. I've moved things around etc.

**Review by oliverlee — APPROVED** [review]

**Pull request merged**